stablecoin-v2/src/hedging_agents.rs

```diff
@@ -0,0 +1,223 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use crate::math::ONE;
+
+#[derive(TypeAbi, TopEncode, TopDecode)]
+pub struct HedgingPosition<M: ManagedTypeApi> {
+    pub collateral_id: TokenIdentifier<M>,
+    pub deposit_amount: BigUint<M>,
+    pub covered_amount: BigUint<M>,
+    pub oracle_value_at_deposit_time: BigUint<M>,
+    pub creation_timestamp: u64,
+    pub withdraw_amount_after_force_close: Option<BigUint<M>>,
+}
+
+#[elrond_wasm::module]
+pub trait HedgingAgentsModule:
+    crate::fees::FeesModule
+    + crate::hedging_token::HedgingTokenModule
+    + crate::math::MathModule
+    + crate::pools::PoolsModule
+    + price_aggregator_proxy::PriceAggregatorModule
+{
+    #[payable("*")]
+    #[endpoint(openHedgingPosition)]
+    fn open_hedging_position(
+        &self,
+        #[payment_token] payment_token: TokenIdentifier,
+        #[payment_amount] payment_amount: BigUint,
+        amount_to_cover: BigUint,
+        max_oracle_value: BigUint,
+    ) -> SCResult<()> {
+        self.require_collateral_in_whitelist(&payment_token)?;
+
+        let collateral_value_in_dollars = self.get_collateral_value_in_dollars(&payment_token)?;
```

Comment:
not sure if intended, but price decimals are not taken into consideration

Reply:
You're right, this calculation is wrong later down the line. Will fix in the next PR, as this part of the code is already slightly changed.